### PR TITLE
Generate integration tests if generating behavioral component

### PR DESCRIPTION
**Pull request by koddsson**
Also renamed the integrations tests to be called system tests.

### CODE REVIEW

**Review by joelhawksley — COMMENTED** [review]
I _believe_ we should call these `system` tests, not `integration` tests. What do you think @manuelpuyol @khiga8?

**Comment by manuelpuyol** [comment]
Yeah I agree, we need to move our existing tests too

File: component_generator.thor [hunk 1]
```diff
@@ -33,6 +33,10 @@ class ComponentGenerator < Thor::Group
     template("templates/test.tt", "test/components/#{underscore_name}_test.rb")
   end
 
+  def create_integration_test
+    template("templates/integration_test.rb.tt", "test/integration/#{underscore_name}_test.rb") if js_package_name
+  end
```
Comment (manuelpuyol) [inline component_generator.thor:38]: we'd also need to generate a preview like https://github.com/primer/view_components/blob/main/demo/test/components/previews/primer/counter_component_preview.rb

**Comment by koddsson** [comment]
> I _believe_ we should call these `system` tests, not `integration` tests. What do you think @manuelpuyol @khiga8?
> Yeah I agree, we need to move our existing tests too

Sounds good to me. Since it wasn't a huge lift, I just renamed the integration directory in this pull request.

File: component_generator.thor [hunk 2]
```diff
@@ -33,6 +33,11 @@ class ComponentGenerator < Thor::Group
     template("templates/test.tt", "test/components/#{underscore_name}_test.rb")
   end
 
+  def create_system_test
+    template("templates/system_test.rb.tt", "test/integration/#{underscore_name}_test.rb") if js_package_name
+    template("templates/system_test_preview.rb.tt", "test/integration/#{underscore_name}_preview.rb") if js_package_name
```
Comment (manuelpuyol) [inline component_generator.thor:38]:
```suggestion
    template("templates/system_test_preview.rb.tt", "test/components/previews/primer/#{underscore_name}_preview.rb") if js_package_name
```

File: component_generator.thor [hunk 3]
```diff
@@ -33,6 +33,11 @@ class ComponentGenerator < Thor::Group
     template("templates/test.tt", "test/components/#{underscore_name}_test.rb")
   end
 
+  def create_system_test
+    template("templates/system_test.rb.tt", "test/integration/#{underscore_name}_test.rb") if js_package_name
```
Comment (manuelpuyol) [inline component_generator.thor:37]:
```suggestion
    template("templates/system_test.rb.tt", "test/system/#{underscore_name}_test.rb") if js_package_name
```

**Pull request merged**